packages/expo-doctor/src/checks/MetroConfigCheck.ts

```diff
@@ -0,0 +1,36 @@
+import { DoctorCheck, DoctorCheckParams, DoctorCheckResult } from './checks.types';
+import { learnMore } from '../utils/TerminalLink';
+import { configExistsAsync, loadConfigAsync } from '../utils/metroConfigLoader';
+
+export class MetroConfigCheck implements DoctorCheck {
+  description = 'Check for issues with metro config';
+
+  sdkVersionRange = '>=46.0.0';
+
+  async runAsync({ projectRoot }: DoctorCheckParams): Promise<DoctorCheckResult> {
+    const issues: string[] = [];
+
+    // configuration check and companion functions adapted from:
+    // https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/project/metroConfig.ts
+    if (await configExistsAsync(projectRoot)) {
+      const metroConfig = await loadConfigAsync(projectRoot);
+      const hasHashAssetFilesPlugin = metroConfig.transformer?.assetPlugins?.find(
```

Comment:
in the future, we probably want to update our check here and in eas cli to do something like what you did with the app config extension work, to check for a symbol that would be included if it was properly extended (rather than just for one possible property like `assetPlugins` here)